tests/test_problem_bank_scripts.py

```diff
@@ -207,7 +207,7 @@ def test_public(paths: dict[str, pathlib.Path], question: str):
             folder = file.parent.name
             outputFolder = outputPath.joinpath(folder)
             assert filecmp.cmp(
-                file, outputFolder.joinpath(file.name), shallow=False
+                file, str(file).replace(str(comparePath), str(outputPath)), shallow=False
```

Comment:
I'd feel more comfortable if we used the built-in `pathlib` module to create these paths if possible (rather than converting to strings and replacing. 

This is a good review: https://realpython.com/python-pathlib/, see specifically [this section](https://realpython.com/python-pathlib/#renaming-files).

Reply:
What you're linking to as far as I can tell is for *renaming* files, as per the [pathlib docs for `Path.rename(target)`](https://docs.python.org/3.11/library/pathlib.html#pathlib.Path.rename):

> Rename this file or directory to the given target, and return a new Path instance pointing to target. On Unix, if target exists and is a file, it will be replaced silently if the user has permission. On Windows, if target exists, [FileExistsError](https://docs.python.org/3.11/library/exceptions.html#FileExistsError) will be raised. target can be either a string or another path object:

  ```python
  >>> p = Path('foo')

  >>> p.open('w').write('some text')
  9
  >>> target = Path('bar')
  >>> p.rename(target)
  PosixPath('bar')
  >>> target.open().read()
  'some text'

  ```
> The target path may be absolute or relative. Relative paths are interpreted relative to the current working directory, not the directory of the Path object.
> It is implemented in terms of [os.rename()](https://docs.python.org/3.11/library/os.html#os.rename) and gives the same guarantees.
> Changed in version 3.8: Added return value, return the new Path instance.

If you would prefer to make sure we have the full path before converting it to a string, we could do the following:

```suggestion
                file, pathlib.Path(str(file.absolute()).replace(str(comparePath.absolute()), str(outputPath))), shallow=False
```

Unless you want something that (ab)uses `pathlib.Path.parent` among other tools whole bunch to attempt pull out the relative fp after the where it diverges into the expected/generated output dirs.

When I initially made this change for the prarielearn tests in #37, I tried really hard to find a better way than string casting the paths to do it, but I couldn't find any tool currently in pathlib to do it, though potentially we'll be able to do it ourselves, or though pathlib/a 3-rd party extension to pathlib whenever this happens: https://discuss.python.org/t/make-pathlib-extensible/3428

Reply:
I suppose another solution thats effectively equivalent to using `str#replace` but is more "native pathlib" is as follows:
```python
>>> import pathlib
>>> path = pathlib.Path("tests/test_question_templates/question_expected_outputs/public/q01_multiple-choice/q01_multiple-choice.md")
>>> path
PosixPath('tests/test_question_templates/question_expected_outputs/public/q01_multiple-choice/q01_multiple-choice.md')
>>> pathlib.Path(*(part if part != "question_expected_outputs" else "question_generated_outputs" for part in path.parts))
PosixPath('tests/test_question_templates/question_generated_outputs/public/q01_multiple-choice/q01_multiple-choice.md')
```
We'd implement that here as such:

```suggestion
                file, pathlib.Path(*(part if part != "question_expected_outputs" else "question_generated_outputs" for part in file.parts)), shallow=False
```

Reply:
Actually I just realized I missed [`pathlib.Path.relative_to(*args)`](https://docs.python.org/3.11/library/pathlib.html#pathlib.PurePath.relative_to) in the docs:
```python
>>> import pathlib
>>> outputDest = pathlib.Path("tests/test_question_templates/question_generated_outputs/")
>>> compareDest =  pathlib.Path("tests/test_question_templates/question_expected_outputs/")
>>> file = compareDest / "public/q01_multiple-choice/q01_multiple-choice.md"
>>> file
PosixPath('tests/test_question_templates/question_expected_outputs/public/q01_multiple-choice/q01_multiple-choice.md')
>>> outputDest / file.relative_to(compareDest)
PosixPath('tests/test_question_templates/question_generated_outputs/public/q01_multiple-choice/q01_multiple-choice.md')
```
which enables the following solution to the paths issue with `pathlib` as natively as it gets:
```suggestion
                file, outputDest / file.relative_to(compareDest), shallow=False
```

Reply:
Perfect! This is a great solution.